library/core/functions.render.php

```diff
@@ -105,32 +105,47 @@ function helpAsset($title, $description) {
      * Handles url-ifying. Adds an optional button or return link.
      *
      * @param string $title The page title.
-     * @param string $buttonText The text appearing on the button.
+     * @param string|array $buttonText The text appearing on the button or an array of button definitions.
      * @param string $buttonUrl The url for the button.
      * @param string|array $buttonAttributes Can be string CSS class or an array of attributes. CSS class defaults to `btn btn-primary`.
      * @param string $returnUrl The url for the return chrevron button.
      * @return string The structured heading string.
      */
     function heading($title, $buttonText = '', $buttonUrl = '', $buttonAttributes = [], $returnUrl = '') {
-
-        if (is_string($buttonAttributes)) {
-            $buttonAttributes = ['class' => $buttonAttributes];
+        if (is_array($buttonText)) {
+            $buttons = $buttonText;
+        } elseif (!empty($buttonText)) {
+            $buttons = [[
+                'text' => $buttonText,
+                'url' => $buttonUrl,
+                'attributes' => $buttonAttributes
+            ]];
+        } else {
+            $buttons = [];
         }
 
-        if ($buttonText !== '') {
-            if (val('class', $buttonAttributes, false) === false) {
-                $buttonAttributes['class'] = 'btn btn-primary';
+        $buttonsString = '';
+        foreach ($buttons as $button) {
+            $buttonText = $button['text'];
+            $buttonUrl = $button['url'];
+            $buttonAttributes = $button['attributes'];
```

Comment:
Currently triggering a notice on the Manage Category page

> Undefined index: attributes

Reply:
Okay I fixed